app/lib/features/chat/widgets/receipts_builder.dart

```diff
@@ -1,18 +1,114 @@
 import 'package:acter/common/providers/chat_providers.dart';
 import 'package:acter/common/themes/app_theme.dart';
 import 'package:acter_avatar/acter_avatar.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart'
+    show Convo, EventSendState;
+import 'package:atlas_icons/atlas_icons.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:quds_popup_menu/quds_popup_menu.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
 
 class ReceiptsBuilder extends ConsumerWidget {
-  final List<String> seenList;
-
+  final Convo convo;
+  final types.Message message;
   const ReceiptsBuilder({
     super.key,
-    required this.seenList,
+    required this.convo,
+    required this.message,
   });
 
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final receipts = message.metadata?['receipts'];
+    EventSendState? sendState = message.metadata?['eventState'];
+    if (receipts != null && receipts.isNotEmpty) {
+      return _UserReceiptsWidget(
+        seenList: (receipts as Map<String, int>).keys.toList(),
+      );
+    } else {
+      if (sendState != null) {
+        switch (sendState.state()) {
+          case 'NotSentYet':
+            return const SizedBox(
+              height: 8,
+              width: 8,
+              child: CircularProgressIndicator(),
+            );
+          case 'SendingFailed':
+            return Row(
+              children: <Widget>[
+                GestureDetector(
+                  onTap: () => _handleCancelRetrySend(),
+                  child: Text(
+                    'Cancel Send',
```

Comment:
Suggestion:
Does `cancel send` button require?

Because most of the messaging app, if message gets failed to send the it keeps showing with retry only action however if user don't want to retry and cancel it then user generally have option for delete message.

Reply:
I just inferred it based on the local echo states, as they include a `Cancelled` state and fn for ```TimelineItem```.i.e. ```cancel_retry()``` which is supposed to be functional?

Reply:
No issue with the current implementation. I am just thinking for `Cancel Send` option needs as user will have delete message option on long-press event.